### PR TITLE
FaceID Auth

### PR DESCRIPTION
**Pull request by akaktus**
[-] Implemented Keychain wrapper for storing login and password.

### CODE REVIEW

File: o-fish-ios/Helpers/KeychainWrapper.swift [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+        return errorIn
+    }
+
+    func readCredentials() -> (error: Error?, credentials: Credentials?) {
+        var errorIn: Error?
+        var credentials: Credentials?
+        do {
+            credentials = try readCredentials(server: server)
+            print("Credentials were read. Login: \(String(describing: credentials?.username)), password \(String(describing: credentials?.password))")
```
Comment (kojiba) [inline o-fish-ios/Helpers/KeychainWrapper.swift:54]: password should not be logged

File: o-fish-ios/Helpers/KeychainWrapper.swift [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        case touch
+        case face
+        case none
+    }
+
+    func addCredentials(_ credentials: Credentials) -> Error? {
+        var errorIn: Error?
+        do {
+            try addCredentials(credentials, server: server)
+            print("Credentials were added. Login: \(credentials.username), password \(credentials.password)")
```
Comment (kojiba) [inline o-fish-ios/Helpers/KeychainWrapper.swift:39]: password should not be logged

File: o-fish-ios/Views/LoginView.swift [hunk 3]
```diff
@@ -45,12 +45,10 @@ struct LoginView: View {
                                     text: self.$username)
                                     .keyboardType(.emailAddress)
 
-                                //TODO need to implement Face id
-                                Button(action: { print("Face id") }) {
-                                    Image(systemName: "faceid")
+                                Button(action: self.getStoredCredentials) {
+                                    Image(systemName: KeychainWrapper.shared.getPictureName())
```
Comment (kojiba) [inline o-fish-ios/Views/LoginView.swift:49]: lets inject this inside the view struct:

```
struct LoginView {
...
   private let keychain = KeychainWrapper.shared
....
```

and use as a 
`self.keychain.getPictureName()` or `keychain.removeCredentials()`, etc

File: o-fish-ios/Views/LoginView.swift [hunk 4]
```diff
@@ -106,11 +106,32 @@ struct LoginView: View {
                 self.errorMessage = "Invalid email or password"
             case .success:
                 print("Logged in")
+
+                if let error = KeychainWrapper.shared.removeCredentials() as? KeychainError {
+                    print(error.localizedDescription)
+                }
+
+                if let error = KeychainWrapper.shared.addCredentials(Credentials(username: username, password: password)) as? KeychainError {
```
Comment (kojiba) [inline o-fish-ios/Views/LoginView.swift:114]: use injected

File: o-fish-ios/Views/LoginView.swift [hunk 5]
```diff
@@ -106,11 +106,32 @@ struct LoginView: View {
                 self.errorMessage = "Invalid email or password"
             case .success:
                 print("Logged in")
+
+                if let error = KeychainWrapper.shared.removeCredentials() as? KeychainError {
```
Comment (kojiba) [inline o-fish-ios/Views/LoginView.swift:110]: use injected

File: o-fish-ios/Views/LoginView.swift [hunk 6]
```diff
@@ -106,11 +106,32 @@ struct LoginView: View {
...
+                }
+
                 self.loggedIn.wrappedValue = true
                 self.presentationMode.wrappedValue.dismiss()
             }
         }
     }
+
+    private func getStoredCredentials() {
+        let data = KeychainWrapper.shared.readCredentials()
```
Comment (kojiba) [inline o-fish-ios/Views/LoginView.swift:125]: use injected

File: o-fish-ios/Helpers/KeychainWrapper.swift [hunk 7]
```diff
@@ -0,0 +1,159 @@
+//
+//  KeychainManager.swift
+//  
+//  Created on 7/23/20.
+//  Copyright © 2020 WildAid. All rights reserved.
+//
+
+import Foundation
```
Comment (am-MongoDB) [inline o-fish-ios/Helpers/KeychainWrapper.swift:8]: Remove

File: o-fish-ios/Helpers/KeychainWrapper.swift [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+            return
+        }
+
+        let access = SecAccessControlCreateWithFlags(nil,
+                                                     kSecAttrAccessibleWhenPasscodeSetThisDeviceOnly,
+                                                     .userPresence,
+                                                     nil) // Ignore any error.
+
+        let context = LAContext()
+        context.touchIDAuthenticationAllowableReuseDuration = 10
```
Comment (am-MongoDB) [inline o-fish-ios/Helpers/KeychainWrapper.swift:90]: Create a constant for this

File: o-fish-ios/Helpers/KeychainWrapper.swift [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+    var localizedDescription: String {
+        return SecCopyErrorMessageString(status, nil) as String? ?? "Unknown error."
+    }
+}
+
+class KeychainWrapper {
+
+    static let shared = KeychainWrapper()
+    private let server = Constants.realmAppId
+    private let reuseDuration: TimeInterval = 10 //The duration for which Touch ID authentication reuse is allowable.
```
Comment (kojiba) [inline o-fish-ios/Helpers/KeychainWrapper.swift:27]: should Configurate from Settings

**Pull request merged**